framework/test/mocha/unit/modules/chain/chain.js

```diff
@@ -329,4 +334,117 @@ describe('Chain', () => {
 			return expect(stubs.modules.module2.cleanup).to.have.been.called;
 		});
 	});
+
+	describe('#_startForging', () => {
+		beforeEach(async () => {
+			await chain.bootstrap();
+			sinonSandbox.stub(chain.forger, 'loadDelegates');
+			sinonSandbox.stub(chain.forger, 'delegatesEnabled').returns(true);
+			sinonSandbox.stub(chain.forger, 'forge');
+			sinonSandbox.stub(chain.loader, 'syncing').returns(false);
+			sinonSandbox.stub(chain.rounds, 'ticking').returns(false);
+			sinonSandbox.stub(chain.scope.sequence, 'add').callsFake(async fn => {
+				await fn();
+			});
+		});
+
+		it('should load the delegates', async () => {
+			await chain._startForging();
+			expect(chain.forger.loadDelegates).to.be.called;
+		});
+
+		describe('in sequence', () => {
+			it('should register a task in Jobs Queue named "nextForge" and a designated interval', async () => {
+				await chain._startForging();
+				expect(stubs.jobsQueue.register).to.be.calledWith('nextForge');
+			});
+
+			it('should halt if no delegates are enabled', async () => {
```

Comment:
I think these tests that contain `let done;` and `await workerPromise` is following really bad pattern. We shouldn't have to rely on this kind of tests. Because in several tests we'd need to arrange that `workerPromise` which is redundant for the things we want to check.

I suggest to extract `nextForge` job to a separate function. Write unit tests for that function. And when it's necessary we can just check if `stubs.jobsQueue.register` was called with correct arguments.

Reply:
@shuse2 @yatki I have created this issue #3958 so this PR can be merged and in order to apply the above suggestion.